src/azure-cli/azure/cli/command_modules/ams/_params.py

```diff
@@ -56,12 +61,13 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
 
     with self.argument_context('ams account') as c:
         c.argument('location', arg_type=get_location_type(self.cli_ctx),
-                   validator=get_default_location_from_resource_group)
+                   validator=get_default_location_from_resource_group, required=False)
         c.argument('tags', arg_type=tags_type)
 
     with self.argument_context('ams account create') as c:
         c.argument('storage_account', storage_account_arg_type,
                    help='The name or resource ID of the primary storage account to attach to the Azure Media Services account. The storage account MUST be in the same Azure subscription as the Media Services account. It is strongly recommended that the storage account be in the same resource group as the Media Services account. Blob only accounts are not allowed as primary.')
+        c.argument('assign_identity', arg_type=get_enum_type(get_managed_identity_allowed_values_list()), help='Set the managed identity on the media services account.')
```

Comment:
I mean you have `assign_identity=False` in the implementation function:
https://github.com/Azure/azure-cli/blob/f3f24bf7989df3c75542bb0976b52b5276d32e94/src/azure-cli/azure/cli/command_modules/ams/operations/account.py#L20-L21

But here you have `get_enum_type` for `assign_identity`, they need to be aligned. 

I would recommend the following considering the possibility of adding `--identity-user-assigned <user ids>` in the future.
```suggestion
        c.argument('assign_identity', options_list=['--identity-system-assigned'], action='store_true', help='Set the system managed identity on the media services account.')
```

Reply:
Thanks for the clarification! I went ahead and split the parameter into system and will eventually add user later this year. I also had to shorten `identity-system-assigned` to `id-system-assigned` to get past the `option_length_too_long` style check.